### ISSUE TITLE
st_read() wrong error message

### ISSUE THREAD

**Issue by emileten** [r-spatial/sf#1326]
`st_read()` throws : 

`Error in st_sf(x, ..., agr = agr, sf_column_name = sf_column_name) : 
  no simple features geometry column present`

when user gives an argument that is not in the function's set of arguments. For example, I wrote `st_read(....., quietly=TRUE)` while the true name of the argument is `quiet`, and got that misleading error message. I thought for a while that my shapefile was wrong, but it was about the argument name.

**REPREX**  (add your path string to download the shapefile) : 

```
library(sf)
# load my shapefile zip file & unzip
    dir = "DIRECTORY_HERE"
    zip_path = file.path(dir, "arrondissements.zip")
    download.file("https://www.data.gouv.fr/en/datasets/r/0e6bac43-1e3f-4fa7-97a3-00fa483fde0a", zip_path)
    unzip(zip_path, exdir=dir)
    shp_path = file.path(dir, "arrondissements.shp")
 #read the shapefile. This works
    st_read(dsn=shp_path, stringsAsFactors=FALSE,geometry_column="geometry",quiet=TRUE)
 #But this doesn't work and throws misleading error message.
    st_read(dsn=shp_path, stringsAsFactors=FALSE,geometry_column="geometry",quietly=TRUE)

```
I think this should be fixed, if I didn't do anything wrong ! 

Emile

**Comment by edzer** [comment]
This would be tricky to implement, as some arguments need to be passed on through `...`, such as `crs`.

**Comment by DDorch** [comment]
I agree that this issue is annoying. 

I have recently changed `rgdal::readOGR` calls by `sf::st_read` and I forgot to remove the `verbose` argument. I got the same confusing error message and I spent more than one hour to understand my mistake.

During my search I found related issues such as https://gis.stackexchange.com/a/305172

Is there any way to, at least, ignore bad arguments instead of raising an incomprehensible error message?
